src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java

```diff
@@ -1,30 +1,106 @@
+            caseService.createInitialCase(caseCreation);
+            consumer.ack();
+          } catch (final IOException e) {
+            log.with(e)
+                .error(
+                    "Something went wrong while processing message received from PubSub "
+                        + "for case creation notification");
+            consumer.nack();
+          }
+        };
+    Subscriber subscriber = getCaseCreationNotificationSubscriber(receiver);
+    // Start the subscriber.
+    subscriber.startAsync().awaitRunning();
+    log.with(subscriber.getSubscriptionNameString())
+        .info("Listening for case creation notification messages on PubSub-subscription id");
+  }
 
   /**
-   * To process SampleUnitParents read from queue
+   * Provides PubSub subscriber for case creation notification against message receiver
    *
-   * @param caseCreation the java representation of the message body
+   * @param receiver: com.google.cloud.pubsub.v1.MessageReceiver;
+   * @return com.google.cloud.pubsub.v1.Subscriber;
+   */
+  private Subscriber getCaseCreationNotificationSubscriber(MessageReceiver receiver)
+      throws IOException {
+    if (StringUtil.isBlank(System.getenv("PUBSUB_EMULATOR_HOST"))) {
+      log.info("Returning Subscriber for case creation notification");
+      ExecutorProvider executorProvider =
+          InstantiatingExecutorProvider.newBuilder().setExecutorThreadCount(4).build();
+      // `setParallelPullCount` determines how many StreamingPull streams the subscriber will open
+      // to receive message. It defaults to 1. `setExecutorProvider` configures an executor for the
+      // subscriber to process messages. Here, the subscriber is configured to open 2 streams for
+      // receiving messages, each stream creates a new executor with 4 threads to help process the
+      // message callbacks. In total 2x4=8 threads are used for message processing.
+      return Subscriber.newBuilder(getCaseCreationSubscriptionName(), receiver)
+          .setParallelPullCount(2)
+          .setExecutorProvider(executorProvider)
+          .build();
+    } else {
+      log.info("Returning emulator Subscriber");
+      return new PubSubEmulator().getEmulatorSubscriberForCaseCreationNotification(receiver);
+    }
+  }
+
+  /***
+   * Provides subscription name for the case notification subscriber
+   * @return com.google.pubsub.v1.ProjectSubscriptionName
    */
-  @Transactional
-  @ServiceActivator(inputChannel = "caseTransformed", adviceChain = "caseRetryAdvice")
-  public void acceptSampleUnit(SampleUnitParent caseCreation) {
-    log.debug("received CaseCreation Message from queue");
-    caseService.createInitialCase(caseCreation);
+  private ProjectSubscriptionName getCaseCreationSubscriptionName() {
+    String project = appConfig.getGcp().getProject();
+    String subscriptionId = appConfig.getGcp().getCaseNotificationSubscription();
+    log.info(
```

Comment:
Is this worth it?  It logs out basically the same information once it's finished (the line with `log.with(subscriber.getSubscriptionNameString())`)

Reply:
yes my bad. will make the change.

Reply:
Actually it is in a way as one is logging a message when creating the subscription object and other is saying that now I am all set and listening to the pubsub. what you say?